### ISSUE TITLE
Search give correct street but in wrong village 

### ISSUE THREAD

**Issue by janusbenissa** [organicmaps/organicmaps#3218]
When searching for Avenida dels Garcies https://www.openstreetmap.org/way/23909866#map=15/38.6729/0.0792 I got the correct result, but OM tells me the street is in The village "Calp" and not in "Benissa"  Where he really is!  Calp and Benissa villages are next to each other.
![1660982343634](https://user-images.githubusercontent.com/15191462/185735574-357f87f6-b408-47ec-aaa8-69bff8256ecd.jpg)

**Comment by janusbenissa** [comment]
Problem still exist in the latest test version from android 2022.08.23-5

**Comment by biodranik** [comment]
There are many similar issues already reported here. Can you please search and reference them?

**Comment by janusbenissa** [comment]
See also:
https://github.com/organicmaps/organicmaps/issues/2490
https://github.com/organicmaps/organicmaps/issues/857

**Comment by janusbenissa** [comment]
Is somebody still looking after this issue ?
It's a silly mistake in my opinion

**Comment by biodranik** [comment]
It is not so easy to fix. We don't store villages/towns borders at the moment.
@vng can we avoid storing borders by using them only at the generation time, and mark searchable objects instead? Will it be a lesser overhead?

**Comment by vng** [comment]
- We store town's borders (not villages).
- Benissa and Calp are towns.
- This is a bug with borders generation and I'm still working on this.

**Comment by janusbenissa** [comment]
Looks like you solved it in the new beta update.

**Comment by vng** [comment]
Yes for this particular case, but the main (final) fix is still in progress.

**Comment by janusbenissa** [comment]
Still in progress?
